src/components/SingleArticle.jsx

```diff
@@ -1,34 +1,51 @@
 import { useEffect, useState } from "react"
-import { useParams } from "react-router-dom"
-import { getArticleById } from "../api"
-import { Link } from "react-router-dom"
+import { useParams, Link } from "react-router-dom"
 import Collapsible from "./Collapsible"
 import CommentList from "./CommentList"
-import { getArticleComments } from "../api"
+import { incrVoteCount, decrVoteCount, getArticleComments, getArticleById } from "../api"
+
 
 const SingleArticle = () => {
     const [article, setArticle] = useState({})
     const [comments, setComments] = useState([])
+    const [votes, setVotes] = useState(0)
     const {article_id} = useParams()
     
+
     useEffect(() => {
         getArticleById(article_id)
         .then((data) => {
             setArticle(data.article)
         })
-    }, [])
+    }, [votes])
```

Comment:
we dont need to re-request the article when the votes get updated - we should just use this state to display the updated number of votes so this state can be set when you initially request the article to article.votes and then render this instead of rendering article.votes - so when we optimistically vary this, it is equal to the number of votes on the article - we dont need to re-request the article every time